android/authentication.md

```diff
@@ -1,615 +1,704 @@
+// NextActivity is the screen users should see after signing-in
+AWSMobileClient.getInstance().showSignIn(this, NextActivity.class);
 ```
-$ amplify push
+
+In the above code you would have created an Android Activity called `NextActivity` which would automatically be nativated to upon successful sign-up and sign-in. For testing, you can alternatively just use `MainActivity.class` after initializing:
+
+```java
+AWSMobileClient.getInstance().showSignIn(this, NextActivity.class, 
+    SignInUIConfiguration.builder().build(), 
+    new Callback<UserStateDetails>() {
+
+        @Override
+        public void onResult(UserStateDetails userStateDetails) {
+            switch (userStateDetails.getUserState()){
+                case SIGNED_IN:
+                    Log.i("INIT", "logged in!");
+                    break;
+                case SIGNED_OUT:
+                    AWSMobileClient.getInstance().showSignIn(MainActivity.this, MainActivity.class);
+                    break;
+                default:
+                    AWSMobileClient.getInstance().signOut();
+                    break;
+            }
+        }
+
+        @Override
+        public void onError(Exception e) {
+            Log.e("INIT", e.toString());
+        }
+    }
+);
 ```
 
-9. Follow the steps at [Set Up Facebook Login](#set-up-facebook) to connect to your backend from your app.
+The above code also shows an additional Auth API, `signOut()`. For more advanced scenarios, you can call the `AWSMobileClient` APIs, such as for building your own UI or using functionality in different UX of your application lifecycle.
 
-#### Google
+### Customization
 
-**To set up Google sign-in**
+Currently, you can change the following properties of the drop-in UI with the `AWSMobileClient`:
+- Logo: Any Drawable resource supported by ImageView
+- Background Color: Any color Android supported
 
-1. In a terminal window, navigate to the root of your app files and add the auth category to your app. The CLI prompts you for configuration parameters.
+```java
+AWSMobileClient.getInstance().showSignIn(this, NextActivity.class, 
+    SignInUIConfiguration.builder()
+    .logoImage(R.id.logo)
+    .backgroundColor(R.color.black)
+    .canCancel(false)
+    .build());
+```
 
-	```
-	$ cd ./YOUR_APP_ROOT
-	$ amplify add auth
-	```
+You can allow the sign in process to be dismissed by setting the `canCancel` property. 
 
-2. Choose to set up your own configuration.
+## Working with the API
 
-	```
-	❯ No, I will set up my own configuration.
-	```
+### SignUp
 
-3. Choose to set up authentication flow using AWS IAM access controls.
+Creates a new user in your User Pool:
 
-	```
-	❯ User Sign-Up, Sign-In, connected with AWS IAM controls ...
-	```
+```java
+final String username = getInput(R.id.signUpUsername);
+final String password = getInput(R.id.signUpPassword);
+final Map<String, String> attributes = new HashMap<>();
+attributes.put("email", "bimin@amazon.com");
```

Comment:
Should we use a generic email address here?

Reply:
Yeah that's a copy/paste miss, fixed